camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelRunner.java

```diff
@@ -232,40 +247,49 @@ private Set<File> resolveFromRepo(List<URI> repositories, Set<String> components
-            // Thus, we will see an error here only if the artifact is resolved by none of the repos in our list.
-            // We could learn to ignore it & let camel throw the error, but our throwing the ResolutionException
-            // seems cleaner.  Notify about the problems closer to the source.
-            Collection<File> resolved = cr.resolve("org.apache.camel", lib, getCamelContext().getVersion(),
-                                                   "Component Resolution");
-            jarSet.addAll(resolved);
-        }
     
-        for (String df : dataformatsToResolve) {
-            String lib = discoverer.findDataFormatForName(df);
-            if (lib == null) {
-                // If we don't know the name, we have no chance of resolving it.  So complain.
-                throw new ResolutionException("Unable to determine jar file for dataformat: " + df);
+            for (String df : dataformatsToResolve) {
+                String lib = discoverer.findDataFormatForName(df);
+                if (lib == null) {
+                    // If we don't know the name, we have no chance of resolving it.  So complain.
+                    throw new ResolutionException("Unable to determine jar file for dataformat: " + df);
+                }
+                // Our resolver is a chain resolver, including all the IBibloResolvers that are in our list of URLs.
+                // Thus, we will see an error here only if the artifact is resolved by none of the repos in our list.
+                // We could learn to ignore it & let camel throw the error, but our throwing the ResolutionException
+                // seems cleaner.  Notify about the problems closer to the source.
+                Collection<File> resolved = cr.resolve("org.apache.camel", lib, getCamelContext().getVersion(),
+                                                       "Dataformat Resolution");
+                jarSet.addAll(resolved);
             }
-            // Our resolver is a chain resolver, including all the IBibloResolvers that are in our list of URLs.
-            // Thus, we will see an error here only if the artifact is resolved by none of the repos in our list.
-            // We could learn to ignore it & let camel throw the error, but our throwing the ResolutionException
-            // seems cleaner.  Notify about the problems closer to the source.
-            Collection<File> resolved = cr.resolve("org.apache.camel", lib, getCamelContext().getVersion(),
-                                                   "Dataformat Resolution");
-            jarSet.addAll(resolved);
         }
         // If our use has specified additional libraries to include, do those now.
         if (additionalLibraries != null && !additionalLibraries.isEmpty()) {
             for (String comp : additionalLibraries) {
                 String[] compParts = comp.split(":");
-                if (compParts.length != 3) {
```

Comment:
Handle cases (normal) where discovered dependencies don't have the camel version.  They don't know it when they're being discovered, so we'll fill it in to match whatever we're running with.